int32 num_balls
---
bool success
